File: Python version/specializations/dlv2/desktop/dlv2_desktop_service.py
from platforms.desktop.desktop_service import DesktopService
from specializations.dlv2.dlv2_answer_sets import DLV2AnswerSets


class DLV2DesktopService(DesktopService):
    """Extention of DesktopService for DLV2."""

    def __init__(self, exe_path):
        super(DLV2DesktopService, self).__init__(exe_path)
        self._load_from_stdin_option = "--"

    def _get_output(self, output, error):
        """Returns a new DLV2AnwerSets from given output and error."""
        return DLV2AnswerSets(output, error)

    def start_async(self, callback, programs, options):
        """Calls start_async method of a superclass."""
        super(DLV2DesktopService, self).start_async(
            callback, programs, options)

    def start_sync(self, programs, options):
        """Calls start_sync method of a superclass and returns its output
        object."""
        return super(DLV2DesktopService, self).start_sync(programs, options)

File: CSharp version/EmbASP/it/unical/mat/embasp/specializations/dlv2/desktop/DLV2DesktopService.cs
using System.Collections.Generic;

namespace it.unical.mat.embasp.specializations.dlv2.desktop
{

	using Callback = it.unical.mat.embasp.@base.Callback;
	using InputProgram = it.unical.mat.embasp.@base.InputProgram;
	using OptionDescriptor = it.unical.mat.embasp.@base.OptionDescriptor;
	using Output = it.unical.mat.embasp.@base.Output;
	using DesktopService = it.unical.mat.embasp.platforms.desktop.DesktopService;

	public class DLV2DesktopService : DesktopService
	{

		private readonly OptionDescriptor competitionOutputOption;

		public DLV2DesktopService(string exe_path) : base(exe_path)
		{
			load_from_STDIN_option = "--stdin";
			competitionOutputOption = new OptionDescriptor("--competition-output");
		}

		protected internal override Output getOutput(string output, string error)
		{
			return new DLV2AnswerSets(output, error);
		}

		public virtual void startAsync(Callback callback, IList<InputProgram> programs, IList<OptionDescriptor> options)
		{
			options.Add(competitionOutputOption);
			base.startAsync(callback, programs, options);
		}

		public override Output startSync(IList<InputProgram> programs, IList<OptionDescriptor> options)
		{
			options.Add(competitionOutputOption);
			return base.startSync(programs, options);
		}
	}

}